clients/drcachesim/simulator/cache_simulator.cpp

```diff
@@ -594,6 +594,48 @@ cache_simulator_t::print_results()
     return true;
 }
 
+int_least64_t
+cache_simulator_t::get_cache_metric(unsigned level, metric_name_t metric, unsigned core,
+                                    cache_split_t split) const
+{
+    caching_device_t *curr_cache;
+
+    if (core >= knobs_.num_cores) {
+        ERRMSG("Wrong core number: %u.\n", core);
+        return 0;
```

Comment:
Xref the top-level thread: I would vote for returning an error.  Could go so far as error enum types; or just return -1 (document that's the error return code) either w/ or w/o the print (could see the print useful if there's no errno; could see it annoying linked into a large system that maybe doesn't know the cache layout and is passing all possible levels and using return code to find the bounds of the layout -- there an errno would be nice).

Ditto below.